### ISSUE TITLE
Bug - pattern comprehension will not accept a single element pattern (pattern without relationship)

### ISSUE THREAD

**Issue by InverseFalcon** [neo4j/neo4j#8758]
This pattern comprehension errors out, but it seems like it should be allowed:

`RETURN [(m:Movie) WHERE m.title CONTAINS 'Matrix' | m]`

The only workaround, when trying to use a pattern without any relationships, is to use a zero-length relationship:

`RETURN [(m:Movie)-[*0]-() WHERE m.title CONTAINS 'Matrix' | m]`

**Comment by Mats-SX** [comment]
The reason this does not work is the same as the reason why this does not work:

```
RETURN (:Movie)
```

The underlying reason is that a single node pattern is not an expression in the Neo4j implementation; the pattern has to have at least one relationship. It's currently debated whether we should lift this (arguably asymmetric) constraint, but there is a real problem that hinders it. Consider this query:

```
RETURN ({name: 'myName'})
```

Does this mean:
- return a list of all nodes that have `name: 'myName'`, or
- return a single literal map with a single property `name: 'myName'`

In other words, there's a syntactic conflict between a parenthesized literal map expression, and a node expression with an inlined property predicate.

I hope this sheds some light on why this isn't doable currently. We know of this issue and are not necessarily happy with the current state, but since workarounds exist, it's not pressingly urgent.

**Comment by hvub** [comment]
As @Mats-SX explained, the behavior is deliberated and adding single element patterns to pattern comprehension would introduce syntax ambiguities. Hence, there is foreseeable way how this will happen.
